[OSDOCS-11001]: Moving HCP disconnected docs (#80675)
openshift/openshift-docs: modules/hcp-agentserviceconfig.adoc

* Registries: The `Registries.conf` field contains information about images and namespaces that need to be consumed from a mirror registry rather than the original source registry.

. Configure the Assisted Service by adding the `AssistedServiceConfig` object, as shown in the following example:
+
[source,yaml]
----
---
apiVersion: agent-install.openshift.io/v1beta1
kind: AgentServiceConfig
metadata:
  annotations:
    unsupported.agent-install.openshift.io/assisted-service-configmap: assisted-service-config <1>
  name: agent
  namespace: multicluster-engine
spec:
  mirrorRegistryRef:
    name: custom-registries <2>
  databaseStorage:
    storageClassName: lvms-vg1
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 10Gi
  filesystemStorage:
    storageClassName: lvms-vg1
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 20Gi
  osImages: <3>
  - cpuArchitecture: x86_64
    openshiftVersion: "4.14"
    rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live-rootfs.x86_64.img <4>
    url: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live.x86_64.iso

>     version: 414.92.202308281054-0

Reviewer: Maybe we can add another sample here to remark that we need to add every OCP release we wanna deploy as HCP by adding that release to this file. Sample:

```
- cpuArchitecture: x86_64
   openshiftVersion: "4.15"
   rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live-rootfs.x86_64.img
   url: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live.x86_64.iso
   version: 415.92.202403270524-0
```

Author: Added

// Module included in the following assemblies:
//
// * hosted_control_planes/hcp-disconnected/hcp-deploy-dc-bm.adoc

:_mod-docs-content-type: PROCEDURE
[id="hcp-agentserviceconfig_{context}"]
= Deploying AgentServiceConfig resources

> The `AgentServiceConfig` custom resource is an essential component of the Assisted Service add-on that is part of {mce-short}. It is responsible for bare metal cluster deployment. When the add-on is enabled, you deploy the `AgentServiceConfig` resource to configure the add-on.

Reviewer 2: ```suggestion
The `AgentServiceConfig` custom resource is an essential component of the Assisted Service add-on that is part of the {mce-short}. It is responsible for bare metal cluster deployment. When the add-on is enabled, you deploy the `AgentServiceConfig` resource to configure the add-on.
```

> In addition to configuring the `AgentServiceConfig` resource, you need to include additional config maps to ensure that {mce-short} functions properly in a disconnected environment.

Reviewer 2: ```suggestion
In addition to configuring the `AgentServiceConfig` resource, you need to include additional config maps to ensure that the {mce-short} functions properly in a disconnected environment.
```

<3> The `spec.osImages` field contains different versions available for deployment by this Operator. This field is mandatory. This example assumes that you already downloaded the `RootFS` and `LiveISO` files.
<4> Add a `cpuArchitecture` subsection for every {product-title} release that you want to deploy. In this example, `cpuArchitecture` subsections are included for 4.14 and 4.15.
<5> In the `rootFSUrl` and `url` fields, replace `dns.base.domain.name` with the DNS base domain name.

. Deploy all of the objects by concatenating them into a single file and applying them to the management cluster. To do so, enter the following command:
+
[source,terminal]
----
$ oc apply -f agentServiceConfig.yaml
----
+
The command triggers two pods.
+
.Example output
[source,terminal]
----
assisted-image-service-0                               1/1     Running   2             11d <1>
assisted-service-668b49548-9m7xw                       2/2     Running   5             11d <2>
----
+

> <1> The `assisted-image-service` pod is responsible for creating the Red Hat Enterprise Linux CoreOS (RHCOS) boot image template, which is customized for each cluster that you deploy.

Reviewer 2: ```suggestion
<1> The `assisted-image-service` pod is responsible for creating the {op-system-first} boot image template, which is customized for each cluster that you deploy.
```
